File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        unordered_map<int,int> lost;
        unordered_set<int>  st;
        for(auto i: matches){
            lost[i[1]]++;
            st.insert(i[0]);
            st.insert(i[1]);
        }
        vector<vector<int>> ans(2);
        for(auto i: st){
            if(lost[i]==0) ans[0].push_back(i);
            else if(lost[i]==1) ans[1].push_back(i);
        }
        sort(ans[0].begin(), ans[0].end());
        sort(ans[1].begin(), ans[1].end());
        return ans;
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        HashMap<Integer, Integer> loseOrTaken= new HashMap<>();
        for(int i=0; i<matches.length; i++){
            loseOrTaken.put(matches[i][1], loseOrTaken.getOrDefault(matches[i][1], 0)+1);
        }
        List<List<Integer>> res = new ArrayList<>();
        res.add(new ArrayList<>());
        res.add(new ArrayList<>());
        for(int i=0; i<matches.length; i++){
            if(loseOrTaken.get(matches[i][1]) ==1) res.get(1).add(matches[i][1]);
            if(loseOrTaken.containsKey(matches[i][0])==false){
                res.get(0).add(matches[i][0]);
                loseOrTaken.put(matches[i][0], -1);
            }
        }
        Collections.sort(res.get(0));
        Collections.sort(res.get(1));
        return res;
    }
}